src/lockfile/mod.rs

```diff
@@ -117,6 +117,22 @@ impl Lockfile {
         self.pkg_specs == cfg.contents.packages && self.global_key_specs == cfg.contents.gpgkeys
     }
 
+    /// Returns true if the lockfile is compatible with the specified configuration
+    /// and if all required rpm packages are included in the lockfile.
+    pub fn all_deps_compatible(&self, cfg: &Config) -> bool {
+        let required_pkgs = match Self::resolve(
```

Comment:
it suffices to check that the requires of local packages are unchanged here

Reply:
Done